src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java

```diff
@@ -0,0 +1,131 @@
+        new String[] {
+            "Home",
+            "Contact form",
+            "Service",
+            "Metals & Colors",
+            "Elements packs"};
+
+    @Test
+    public void testLoginAndGeneralHomePageElements() {
+        // set up the soft assertions
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        // 1) Open test site by URL
+        driver.get(url);
+
+        // 2) Assert Browser title
+        softAssertions.assertThat(driver.getTitle()).isEqualTo("Home Page");
+
+        // 3) Perform login
+        WebElement loginForm = driver.findElement(By.id("user-icon"));
+        loginForm.click();
+
+        WebElement name = driver.findElement(By.id("name"));
+        name.sendKeys(username);
+
+        WebElement password = driver.findElement(By.id("password"));
+        password.sendKeys(pass);
+
+        WebElement loginButton = driver.findElement(By.id("login-button"));
+        loginButton.click();
+
+        // 4) Assert Username is logged
+        WebDriverWait webDriverWait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        WebElement userFirstSecondName = driver.findElement(By.id("user-name"));
+
+        // wait until the username is no more hidden
+        webDriverWait.until(ExpectedConditions.visibilityOf(userFirstSecondName));
+
+        softAssertions.assertThat(userFirstSecondName.getText()).isEqualTo(usernameInfo);
+
+        // 5) Assert that there are 4 items on the header -------------------------------------
+        // section are displayed and they have proper texts
+        List<WebElement> navBarItems = driver.findElements(By.xpath("//nav/ul/li/a[not(@href='#')]"));
+        assertItemsCountPlacementAndTexts(navBarItems, 4, expectedNavBarTitles, softAssertions);
+
+        // 6) Assert that there are 4 images on the
+        // Index Page and they are displayed
+        List<WebElement> icons = driver.findElements(By.className("icons-benefit"));
+        softAssertions.assertThat(icons.size()).isEqualTo(4);
+        icons.forEach(icon -> softAssertions.assertThat(icon.isDisplayed()).isTrue());
+
+        // 7) Assert that there are 4 texts on the
+        // Index Page under icons and they have proper text
+        List<WebElement> textsUnderImages = driver.findElements(By.className("benefit-txt"));
+        assertItemsCountPlacementAndTexts(textsUnderImages, 4, expectedTextsUnderIcons, softAssertions);
+
+        // 8) Assert that there is the iframe with “Frame Button” exist
+        WebElement frame = driver.findElement(By.id("frame"));
+        softAssertions.assertThat(frame).isNotNull();
+
+        // 9) Switch to the iframe and check that there is “Frame Button” in the iframe
+
+        // remembering of the main window
+        String windowHandle = driver.getWindowHandle();
+
+        driver.switchTo().frame("frame");
+        WebElement frameButton = driver.findElement(By.id("frame-button"));
+        softAssertions.assertThat(frameButton).isNotNull();
```

Comment:
Since plain `findElement` cannot return `null` I suggest you replace this assertion by some other assertion, for example `element.isDisplayed()` and `element.isEnabled()`